website/public/js/controllers/inventoryCtrl.js

```diff
@@ -112,6 +115,10 @@ habitrpg.controller("InventoryCtrl",
       var potName = Content.hatchingPotions[potion.key].text();
       if (!$window.confirm(window.env.t('hatchAPot', {potion: potName, egg: eggName}))) return;
       user.ops.hatch({params:{egg:egg.key, hatchingPotion:potion.key}});
+      if (!user.preferences.suppressModals.hatchPet) {
+        $rootScope.hatchedPet = {egg: eggName, potion: potName, eggKey: egg.key, pet: 'Pet-' + egg.key + '-' + potion.key};
```

Comment:
It's not your fault. The app has been in this state for a while now. Bad code begets bad code...